### PR TITLE
🔒 Update supported TLS cipher suites

### PR DESCRIPTION
**Pull request by canterberry**
The list of cipher suites included in this commit are consistent with the values and precedence in the [Golang TLS documentation](https://golang.org/src/crypto/tls/cipher_suites.go).

## Removed

No cipher suites were removed in this PR. For existing cipher suites, only the order has changed.

## Added

The following cipher suites were added in this PR:

 * `TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256`
 * `TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305`
 * `TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256`
 * `TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305`
 * `TLS_RSA_WITH_AES_128_CBC_SHA256`

## Security Note

Cipher suites with RC4 are still included within the list of accepted values for compatibility, but **these cipher suites are not safe to use** and should be deprecated with warnings and subsequently removed. Support for RC4 ciphers has already been removed or disabled by default in many prominent browsers and tools, including Golang.

### References

 * [RC4 on Wikipedia](https://en.wikipedia.org/wiki/RC4)
 * [Mozilla Security Blog](https://blog.mozilla.org/security/2015/09/11/deprecating-the-rc4-cipher/)

### CODE REVIEW

**Comment by pearkes** [comment]
Thanks for the PR @canterberry!

Prior to review it'd be worth investigating out the source of the test failures here.

```
--- FAIL: TestFullConfig (0.04s)
    runtime_test.go:3624: "*RuntimeConfig.AdvertiseAddrLAN.IP[0]" is zero value
--- FAIL: TestFullConfig/json (0.02s)
    values.go:22: verification for runtime_config at runtime_test.go:3658:
```

You can run tests locally with `make test` or view the CI failures linked in this PR.

**Comment by canterberry** [comment]
Thanks for the review @pearkes! At the time I forked for this PR, the `master` branch was failing. I'm fairly confident that is the cause of this PR's failure as well. It looks like `master` is passing now, so I'll merge that in now.

**Comment by canterberry** [comment]
@pearkes Fixed. The failing tests were due to differences in the three different configuration definitions in the test suite, which all need to be in sync for the expectations to work out properly. The error reported by the failing test is misleading -- it seems to occur if any definition within the JSON or HCL payloads do not match up with the corresponding values in the `RuntimeConfig` definition.

The most recent failing Travis build appears to be sporadic. Local builds are failing for me due to the HTTP test suite creating too many open files, but the tests shown by Travis CI as failing do pass for me locally.

**Comment by canterberry** [comment]
(squashing latest two commits to coerce a Travis CI rebuild)

File: agent/consul/server_test.go [hunk 1]
```diff
@@ -767,10 +767,10 @@ func TestServer_RevokeLeadershipIdempotent(t *testing.T) {
 	dir1, s1 := testServer(t)
 	defer os.RemoveAll(dir1)
 	defer s1.Shutdown()
-	
+
 	testrpc.WaitForLeader(t, s1.RPC, "dc1")
 
-	err:= s1.revokeLeadership()
+	err := s1.revokeLeadership()
```
Comment (canterberry) [inline agent/consul/server_test.go]: **Note:** The seemingly spurious changes in this file result from changes applied by `make format`.

**Comment by banks** [comment]
Thanks @canterberry!

**Pull request merged**